.github/workflows/docker.yaml

```diff
@@ -20,18 +20,26 @@ jobs:
       IMAGE: ghcr.io/ros-planning/moveit2:${{ matrix.ROS_DISTRO }}-${{ github.job }}
 
     steps:
-      - uses: actions/checkout@v2
-
-      - name: Build image
-        run: |
-          echo Building docker ${{ env.IMAGE }}
-          docker build . --build-arg ROS_DISTRO=${{ matrix.ROS_DISTRO }} --file .docker/${{ github.job }}/Dockerfile --tag ${{ env.IMAGE }}
-
-      - name: Log in to registry
-        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
-
-      - name: Push image
-        run: docker push ${{ env.IMAGE }}
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Login to Github Containter Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build and Push
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          file: .docker/${{ github.job }}/Dockerfile
+          build-args: ROS_DISTRO=${{ matrix.ROS_DISTRO }}
+          pull: true
+          push: true
+          no-cache: false
```

Comment:
What's the purpose of `pull` and `no-cache`?
I couldn't find any info about these attributes in the [docs](https://github.com/docker/build-push-action/blob/master/docs/advanced/cache.md).

Reply:
It's on the main README: https://github.com/docker/build-push-action.

`pull`: always tries to pull a new image
`no-cache`: if true don't use cache. This defaults to false though, I can actually remove this, put it here for a little more verbosity on what I was doing.

Reply:
Do you really need to pull the newest image if you are going to rebuild it anyway?
Yes, I suggest removing `no-cache`. Keep the config as simple as possible.